libs/net/services/Actions/IngestAction.cs

```diff
@@ -34,7 +34,7 @@ public IngestAction(IApiService api, IOptions<TOptions> options) : base(api, opt
     /// <param name="manager"></param>
     /// <param name="name"></param>
     /// <param name="cancellationToken"></param>
-    public abstract Task PerformActionAsync(IDataSourceIngestManager manager, string? name = null, CancellationToken cancellationToken = default);
+    public abstract Task PerformActionAsync<T>(IDataSourceIngestManager manager, string? name = null, T? data = null, CancellationToken cancellationToken = default) where T : class;
```

Comment:
This isn't required presently, but allows us to pass generic data when making calling the action.